app/controllers/collection_show_controller.rb

```diff
@@ -43,7 +43,20 @@ def facet
     end
     @facet = blacklight_config.facet_fields[params[:id]]
 
-    @response = search_service.facet_field_response(@facet.key)
+    begin
+      @response = search_service.facet_field_response(@facet.key)
+    rescue ActionController::UnpermittedParameters
+      # Blacklight will raise an UnpermittedParameters error here if
+      # a bot tries to pass an array into the facet.page param.
+      if params["facet.page"].is_a? String
+         # Reraise.
+        raise
+      else
+        # Quietly return an :unprocessable_entity.
+        return render plain: "Error: unpermitted parameters.", status: :unprocessable_entity
+      end
+    end
+
```

Comment:
I am not opposed to rescueing this error, but if so we should do it using the Rails controller `rescue_from` feature we talked about earlier, rather than our own begin/rescue/end block. 

I also do not understand the if/else going on there.  While the comment suggests the first branch is about when an Array is being passed into `facet.page`, of course the check `params["facet.page"].is_a? String` will NOT be true if an array was passed into facet.page, will it? 

If the intention is to rescue if and only if `! params["facet.page"].is_a? String` (that's what it does I think?), we could maybe use more comments explaining why -- or if this condition really is somehow met only when it was an Array... I don't understand how that is possible so comments explaining that?

I also think it might be defensible to just rescue all `ActionController::UnpermittedParameters`

Reply:
Using `rescue_from` now, and better comments.